stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/resiliency/failover-connector.bal

```diff
@@ -0,0 +1,523 @@
+                }
+            }
+            if (response != null) {
+                int statusCode = response.statusCode;
+                if (retryErrorCodes[statusCode] == true) {
+                    if (arrayLength > runningIndex) {
+                        sleep(retryInterval);
+                        failOverClient = failOverClientsArray[runningIndex];
+                        bind failOverClient with endPoint;
+                    } else {
+                        return populateFailoverErrorFromLastResponse(response, failoverError, runningIndex - 1);
+                    }
+                } else {
+                    break;
+                }
+            }
+        }
+        return response, null;
+    }
+
+    @Description {value:"The GET action implementation of the Failover Connector."}
+    @Param {value:"path: Resource path"}
+    @Param {value:"request: An OutRequest struct"}
+    @Return {value:"The InResponse struct"}
+    @Return {value:"Error occurred during the action invocation, if any"}
+    action get (string path, http:OutRequest request) (http:InResponse, http:HttpConnectorError) {
+
+        int arrayLength = lengthof (failOverClientsArray);
+        int runningIndex = 0;
+        http:HttpClient failOverClient = failOverClientsArray[runningIndex];
+        bind failOverClient with endPoint;
+        http:InResponse response;
+        http:HttpConnectorError err = {};
+        FailoverConnectorError failoverError = {};
+        failoverError.httpError = [];
+        int initialIndex = runningIndex;
+        int startingIndex = -1;
+
+        while (startingIndex != runningIndex) {
+            startingIndex = initialIndex;
+            runningIndex = runningIndex + 1;
+            response, err = endPoint.get(path, request);
+            if (response == null && err != null) {
+                if (arrayLength > runningIndex) {
+                    sleep(retryInterval);
+                    failoverError.httpError[runningIndex - 1] = err;
+                    failOverClient = failOverClientsArray[runningIndex];
+                    bind failOverClient with endPoint;
+                } else {
+                    return populateGenericFailoverError(failoverError, err, runningIndex - 1);
+                }
+            }
+            if (response != null) {
+                int statusCode = response.statusCode;
+                if (retryErrorCodes[statusCode] == true) {
+                    if (arrayLength > runningIndex) {
+                        sleep(retryInterval);
+                        failOverClient = failOverClientsArray[runningIndex];
+                        bind failOverClient with endPoint;
+                    } else {
+                        return populateFailoverErrorFromLastResponse(response, failoverError, runningIndex - 1);
+                    }
+                } else {
+                    break;
+                }
+            }
+        }
+        return response, null;
+    }
+}
+
```

Comment:
It seems bulk of the code is repeating in each action. Shall we figure out a way to solve that. 

One thing I can think of is having common function which takes the connector action as an argument. Then move the rest of variables to the top of the connector.